itests/bin/generate-cli-splits.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+usage() {
+	echo "$0 <from> <to>"
+	exit 1
+}
+
+[ "$1" == "" ] && usage
+[ "$2" == "" ] && usage
+
+
+inDir="$1"
+outDir="$2"
+
+git grep SplitSupport.process | grep "$1" | cut -d ':' -f1|while read f;do
```

Comment:
Nit: Please be consistent about putting two spaces next to the | characters when they are used as pipeline separators.

Reply:
As I see in this model the number of batches would be fixed, not the number of tests in a batch. I see why is this useful for generating the files, but it would be better to fix the batch size.

How the isolated tests are going to be handled?

Reply:
yes the number of batches will be fixed; and it will have nothing to do with the number of batches in a test suite.

I've considered a highly sophisticated one prior to ending up with this solution (which was also inherently complex as well)...after some time I concluded that actually there is no big gain from making it "very good"...if all testclasses will be below say 30 minutes it will be good enough.

So the goal of this thing is not to make the perfect sized split; but to make them run under say N minutes; when I've calibrated the actual split counts I tried to go for ~15 minutes so it will take some time when it will needs adjusting from time-to-time.

About isolated tests: this patch doesn't tries to address that - right now I think the option to put tests into "isolation" will not be supported anymore. The best would be to fix them but putting them on ignore could be also an option...
Right now I don't see any easy way to define "isolated" suites...however....it could be probably forced into this splits logic in some way...for now I think let's fix or ignore them - later we can get back to this